IOPool/Output/src/RootOutputFile.cc

```diff
@@ -196,7 +197,15 @@ namespace edm {
     parentageTree_ = RootOutputTree::makeTTree(filePtr_.get(), poolNames::parentageTreeName(), 0);
     parameterSetsTree_ = RootOutputTree::makeTTree(filePtr_.get(), poolNames::parameterSetsTreeName(), 0);
 
-    fid_ = FileID(createGlobalIdentifier());
+    if (overrideGUID.empty()) {
```

Comment:
Doesn't PoolOutputModule support writing multiple files (in the condition a file gets too large)? In that case, how does this override interact with it?

Reply:
I wondered about that case too. With this PR both files would get the same GUID (from the configuration). That doesn't sound very good.

I can think of some alternatives:
* first file gets the GUID from configuration, and subsequent files get a generated GUID
* prevent multiple output files per PoolOutputModule instance (e.g. throw an exception if second file is opened)

But those don't sound very attractive either. Maybe subsequent files getting a generated GUID would be the least bad option? Or would you have more ideas?

Reply:
The last commit implements the 

> * first file gets the GUID from configuration, and subsequent files get a generated GUID

approach